### DISCUSSION TITLE
Introduce a new macro to the core, well not really - search-filter-count?

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#7203] [Ideas]
I would like to introduce a new macro to the core? "search-filter-count" but in fact it is already in the core, just not as a global macro and without any documentation.  I am seeking support to submit this as a core change.
* I will assume for a moment that the reason for this is self evident but please ask if you need more justification?
* It would not add more than a few dozen bytes as it is mostly code reuse.
* I do plan to learn how to submit PR's to the core but until then could someone act on my behalf?
* I am, also seeking peer review here.

It involves
* A new macro tiddler `$:/core/macros/search-filter` (see **Proposed new tiddler $:/core/macros/search-filter** below)
* Editing `$:/core/ui/ControlPanel/Basics` to remove the redundant code

Then the documentation
* Subsequently documenting the two new macros with examples 

Background

Look inside <kbd>Control Panel > Info > Basics </kbd> and to the bottom, you see;
![Snag_e4a255](https://user-images.githubusercontent.com/31584658/211958771-4f270fc5-5ab7-40f4-823d-557d0c3784a5.png)
* This happens to be in the tiddler $:/core/ui/ControlPanel/Basics

The macro is defined internally as;
 ```
\define show-filter-count(filter)
\whitespace trim
<$button class="tc-btn-invisible">
<$action-setfield $tiddler="$:/temp/advancedsearch" $value="""$filter$"""/>
<$action-setfield $tiddler="$:/temp/advancedsearch/input" $value="""$filter$"""/>
<$action-setfield $tiddler="$:/temp/advancedsearch/refresh" text="yes"/>
<$action-setfield $tiddler="$:/state/tab--1498284803" $value="$:/core/ui/AdvancedSearch/Filter"/>
<$action-navigate $to="$:/AdvancedSearch"/>
<$action-sendmessage $message="tm-focus-selector" $param=".tc-advanced-search input"/>
''<$count filter="""$filter$"""/>''
&#32;
{{$:/core/images/advanced-search-button}}
</$button>
\end
```
* The above could simply be placed in a tiddler such as `$:/core/macros/search-filter`, with the `$:/tags/Macro` and it would be available to all
   * and we could remove the definition in `$:/core/ui/ControlPanel/Basics` recovering most of the bytes needed
   * we would then separately document the macros use.

However a minor variation will provide a second macro showing the search icon with or without the count

### Proposed new tiddler  `$:/core/macros/search-filter` 
```
\define show-filter-count(filter)
<<search-filter-button "$filter$" show-count:"yes">>
\end
\define search-filter-button(filter show-count:"no")
<$button class="tc-btn-invisible" tooltip="Search with filter '$filter$'">
<$action-setfield $tiddler="$:/temp/advancedsearch" $value="""$filter$"""/>
<$action-setfield $tiddler="$:/temp/advancedsearch/input" $value="""$filter$"""/>
<$action-setfield $tiddler="$:/temp/advancedsearch/refresh" text="yes"/>
<$action-setfield $tiddler="$:/state/tab--1498284803" $value="$:/core/ui/AdvancedSearch/Filter"/>
<$action-navigate $to="$:/AdvancedSearch"/>
<$list filter="[[$show-count$]match[yes]]">''<$count filter="""$filter$"""/>''</$list>
{{$:/core/images/advanced-search-button}}
</$button>
\end
```

**Comment by Jermolene** [comment]
Hi @AnthonyMuscio making `show-filter-count` a global macro is a good idea, and would be a straightforward PR for a new contributor.

**Reply by AnthonyMuscio** [reply]
Determining the next step is now the issue for me. 

Other questions
* Is it a 5.2.6 or 5.3.0 change?
